File: result/repair/transcoder/java2python/python_repair/FIND_THE_ELEMENT_BEFORE_WHICH_ALL_THE_ELEMENTS_ARE_SMALLER_THAN_IT_AND_AFTER_WHICH_ALL_ARE_GREATER_THAN_IT.py
def f_gold ( arr , n ) :
    left_max = [ int ( x ) for x in arr ]
    right_min = int ( arr [ - 1 ] )
    for i in range ( 1 , n ) :
        left_max [ i ] = max ( left_max [ i - 1 ] , arr [ i - 1 ] )
    right_min = sys.maxsize
    for i in range ( n - 1 , - 1 , - 1 ) :
        if (left_max[i]<arr[i] and right_min>arr[i]) :
            return i
        right_min = min ( right_min , arr [ i ] )
    return - 1

File: result/translate/starchat/java2cpp/c++/FIND_THE_ELEMENT_BEFORE_WHICH_ALL_THE_ELEMENTS_ARE_SMALLER_THAN_IT_AND_AFTER_WHICH_ALL_ARE_GREATER_THAN_IT.cpp
int f_gold ( int arr [ ] [ ], int n, int m ) {
  int leftMax [ n ] [ m ] = { { 0 } } ;
  for ( int i = 1 ;
  i < n ;
  i ++ ) {
    for ( int j = 0 ;
    j < m ;
    j ++ ) {
      if ( i == 0 ) leftMax [ i ][ j ] = INT_MIN ;
      else leftMax [ i ][ j ] = max ( leftMax [ i - 1 ][ j ], arr [ i - 1 ][ j ] ) ;
    }
  }
  int rightMin = INT_MAX ;
  for ( int i = n - 1 ;
  i >= 0 ;
  i -- ) {
    for ( int j = 0 ;
    j < m ;
    j ++ ) {
      if ( leftMax [ i ][ j ] < arr [ i ][ j ] && rightMin > arr [ i ][ j ] ) return i ;
      rightMin = min ( rightMin, arr [ i ][ j ] ) ;
    }
  }
  return - 1 ;
}